src/filters/filter_lua_script.cpp

```diff
@@ -0,0 +1,382 @@
+            );
+            lua[names_in_color.at(i)] = color;
+        }
+
+        // execute update script in lua
+//        sol::protected_function_result script_update_res = script_update();
+        try {
+            update_lua();
+        } catch (const std::exception& e) {
+            ::spdlog::warn("Update of lua has failed: {}", e.what());
+            throw filter_config_exception(std::string("update script in lua had an error: ") + e.what());
+        }
+
+//        // optionally, check if it worked
+//        if (!script_update_res.valid()){
+//            sol::error err = script_update_res;
+//            ::spdlog::warn("Output of lua update has failed: {}", err.what());
+//        }
+        // receive output data from lua
+        for (size_t i = 0; i < out_eight_bit.size(); i++) {
+            out_eight_bit.at(i) = std::max(std::min(std::round(lua[names_out_eight_bit.at(i)].get_or((double) out_eight_bit.at(i))), 255.0), 0.0);
+        }
+        for (size_t i = 0; i < out_sixteen_bit.size(); i++) {
+            out_sixteen_bit.at(i) = std::max(std::min(std::round(lua[names_out_sixteen_bit.at(i)].get_or((double) out_sixteen_bit.at(i))), 65535.0), 0.0);
+        }
+        for (size_t i = 0; i < out_float.size(); i++) {
+            out_float.at(i) = lua.get_or(names_out_float.at(i), out_float.at(i));
+        }
+        for (size_t i = 0; i < out_color.size(); i++) {
+            out_color.at(i).hue = lua[names_out_color.at(i)]["h"].get_or(out_color.at(i).hue);
+            out_color.at(i).saturation = lua[names_out_color.at(i)]["s"].get_or(out_color.at(i).saturation);
+            out_color.at(i).iluminance = lua[names_out_color.at(i)]["i"].get_or(out_color.at(i).iluminance);
+        }
+
+        // Going through the table and set output channels
+        sol::object outputs = lua["output"];
+        if (outputs.get_type() == sol::type::table) {
+            for(int universe_id = 0; universe_id <= 2 * ((sol::table) outputs).size(); universe_id++){
+                sol::object universe = ((sol::table) outputs)[universe_id];
+                if(auto uptr = dmxfish::io::get_universe(universe_id); uptr != nullptr) {
+                    if (universe.get_type() == sol::type::table) {
+                        for (uint16_t chan = 0; chan < 512; chan++) {
+                            sol::object channel = ((sol::table) universe)[chan];
+                            if (channel.get_type() == sol::type::number) {
+                                uint8_t value = ((sol::table) universe)[chan];
+                                (*uptr)[chan] = value;
+//                                std::cout << "test : " << (int) value << std::endl;
+//                            } else if (channel.get_type() != sol::type::nil) {
+//                                std::cout << " test : is not nil : " << std::endl;
+                            }
+                        }
+                    }
+                }
```

Comment:
If we have large code blocks doing just one isolated thing we should introduce an inlined method for that.

Reply:
moved